sklearn/neighbors/nearest_centroid.py

```diff
@@ -177,6 +177,10 @@ def _partial_fit(self, X, y, classes=None, _refit=False):
             # Number of clusters in each class.
```

Comment:
As a user, I find the warning on lines 214-217 to be quite annoying. This code was present in the original implementation, but it might make sense to move it to the "init" instead so the user only gets the warning once. Or we could move it to the "if _refit or _check_partial_fit_first_call(self, classes)" conditional on line 173. Previously, we only received the warning when calling "fit", which isn't too often. But "partial_fit" is meant to be called often on small batches of data, and this warning gets printed every time, which is redundant and tiresome.

Reply:
I think having it only appear on the first partial_fit is reasonable.